### ISSUE TITLE
JSON API relationships null

### ISSUE THREAD

**Issue by nahtnam** [rails-api/active_model_serializers#1180]
Hey!

I am kind of in a weird situation. I have my models nested under a namespace called `steam` and in one of my relations, I have set: `has_many :status_datacenters, foreign_key: :steam_status_id`.

When I try to render out the JSON API, I get something like this: https://gist.github.com/nahtnam/ee105a6d2ac1634eafdf#file-status-json-L84

Notice how the relationships data is null. It wasnt like that in my old app when I didnt have a custom foreign_key set.

TL;DR: I think setting the foreign_key is throwing JSON API off.

Thanks.

**Comment by beauby** [comment]
This is supposed to happen for an empty to-one relationship. Could you post your models and serializers?

**Comment by nahtnam** [comment]
I'm not on my computer right now but I found a gist I made. https://gist.github.com/nahtnam/890cf453e1b4af4ec008

Ignore the errors and stuff. I fixed it by setting a foreign key on the has_many.

**Comment by beauby** [comment]
@nahtnam Could you post your serializer as well? It seems like you're never calling `render`. Did you include all the related logic?

**Comment by nahtnam** [comment]
Hey!
Here are the serializers and the render:
https://gist.github.com/nahtnam/c1eac377cfb3aec5e51e
Thanks.

On Mon, Sep 21, 2015 at 9:15 AM Lucas Hosseini <notifications@github.com> wrote:
@nahtnam Could you post your serializer as well? It seems like you're never calling render . Did you include all the related logic?

—
Reply to this email directly or view it on GitHub .



Thanks.

**Comment by beauby** [comment]
As far as I know, AMS does not support any option called `foreign_key`, nor should it be aware of the internal DB representation of your data. As long as your model exposes the relationship through the right method, you shouldn't need to specify anything more.
Could you make sure the issue is not happening when removing the `foreign_key` option?

**Comment by nahtnam** [comment]
@beauby Sorry? I dont have `foreign_key` set in the serializers, only in the model. If I remove the `foreign_key` from the rails model, rails throws an error saying that it cannot find the related datacenters. I havent set `foreign_key` on AMS at all.

**Comment by beauby** [comment]
Sorry about that, I read too fast. I'll look into it.

**Comment by beauby** [comment]
There is probably a problem with your models. Could you give the result of `Steam::StatusDatacenter.first.status` inside rails console?

**Comment by beauby** [comment]
Closing this one as it is probably a usage issue. Feel free to reopen if needed.
